server/src/main/java/org/opensearch/index/merge/UnreferencedFileCleanUpStats.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.merge;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * Stores stats about unreferenced file cleanup due to segment merge failure.
+ *
+ * @opensearch.internal
+ */
+public class UnreferencedFileCleanUpStats implements Writeable, ToXContentFragment {
+
+    private long totalUnreferencedFileCleanupCount;
+
+    public UnreferencedFileCleanUpStats() {
+
+    }
+
+    public UnreferencedFileCleanUpStats(StreamInput in) throws IOException {
+        totalUnreferencedFileCleanupCount = in.readVLong();
+    }
+
+    public void add(long totalUnreferencedFileCleanupCount) {
+        this.totalUnreferencedFileCleanupCount += totalUnreferencedFileCleanupCount;
+    }
+
+    public long getTotalUnreferencedFileCleanupCount() {
+        return totalUnreferencedFileCleanupCount;
+    }
+
+    public void add(UnreferencedFileCleanUpStats cleanUpStats) {
+        if (cleanUpStats == null) {
+            return;
+        }
+
+        this.totalUnreferencedFileCleanupCount += cleanUpStats.totalUnreferencedFileCleanupCount;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(Fields.UNREFERENCED_FILE_CLEANUP);
+        builder.field(Fields.COUNT, totalUnreferencedFileCleanupCount);
```

Comment:
Is there a way to clarify whether this denotes the number of times the cleanup was triggered or if this denotes the number of files that were cleaned up?

Reply:
This denotes the number of times the cleanup was triggered.

Reply:
Does using `number_of_executions` instead of `count` seem better? I can't think of a better subfield name as of now. Let me know if you have another name in mind. 

My only concern with using `count` is that if we introduce similar subfields in future, lets say number of files cleaned up or something similar, the `COUNT` subfield name can become ambiguous from a UX perspective.

Reply:
Modified stats name to ```total_cleanup_performed```. Addressed the comments.